src/acom_music_box/music_box_main.py

```diff
@@ -0,0 +1,58 @@
+from acom_music_box import MusicBox
+from acom_music_box import music_box_logger
+
+
+import math
+import datetime
+import sys
+
+
+
+# Retrieve named arguments from the command line and
+# return in a dictionary of keywords.
+# argPairs = list of arguments, probably from sys.argv
+#       named arguments are formatted like this=3.14159
+# return dictionary of keywords and values
+def getArgsDictionary(argPairs):
+   argDict = {}
+
+   for argPair in argPairs:
+      # the arguments are: arg=value
+      pairValue = argPair.split('=')
+      if (len(pairValue) < 2):
+         argDict[pairValue[0]] = None
+         continue
+
+      argDict[pairValue[0]] = pairValue[1]
+
+   return(argDict)
```

Comment:
Should we consider using [`argparse`](https://docs.python.org/3/library/argparse.html) instead?

Reply:
@K20shores I investigated the python argparse module. It does not handle named arguments very well. You have to do this:
myExec -something 3.14159 -else 2.71828
instead of this:
myExec something=3.14159 else=2.71828

argparse is more suited to abbreviated command lines, as in:
tar -cvf myfolder.tar myfolder

For a long series of named arguments, I think it would become difficult to scan them all in a batch file and make sure the pairs match up. The named arguments are also similar to the CGI format, like this:

https://www.acom.ucar.edu/cgi-bin/acom/tuv-x.py?wavelength=0&wStart=280&wEnd=700&wIntervals=420&inputMode=1&latitude=40&longitude=-105&date=20240320&timeStamp=19%3A00%3A00&zenith=0&ozone=300&albedo=0.1&gElevation=0&height=0&temperature=290&density=2.5e%2B19&directSun=1.0&diffDown=1.0&diffUp=1.0&time=19&nStreams=2&tauCloud=0.25&ztop=5.00&cloudAlbedo=0.9999&cloudAsymmetry=0.85&zbase=4.00&tauAer=0.235&aerosolLambda0=550&ssaAer=0.990&aerosolAsymmetry=0.62&aerosolAlpha=1.000

Let's talk about this at our 2:30 PM. I have invited @mattldawson to our meeting as well.

Reply:
ChatGPT seems to think we could do this


```
import argparse

class KeyValueAction(argparse.Action):
    def __call__(self, parser, namespace, values, option_string=None):
        for value in values:
            key, val = value.split('=')
            setattr(namespace, key, val)

parser = argparse.ArgumentParser(description='Process some key=value pairs.')

parser.add_argument(
    'key_value_pairs',
    nargs='+',  # This means one or more arguments are expected
    action=KeyValueAction,
    help='Arguments in key=value format'
)

args = parser.parse_args()
print(vars(args))
```

Reply:
I replaced the innards of `def getArgsDictionary(argPairs):` with the ChatGPT code, mostly intact.